src/ntfs/write.rs

```diff
@@ -179,6 +177,229 @@ pub fn write_fares_v1(
+    tickets: &CollectionWithId<Ticket>,
+    ticket_uses: &CollectionWithId<TicketUse>,
+    ticket_prices: &Collection<TicketPrice>,
+    ticket_use_perimeters: &Collection<TicketUsePerimeter>,
+    ticket_use_restrictions: &Collection<TicketUseRestriction>,
+) -> Result<()> {
+    fn has_no_transfer(ticket_uses: &CollectionWithId<TicketUse>, ticket_use_id: &String) -> bool {
+        if let Some(ticket_use) = ticket_uses.get(ticket_use_id) {
+            if (ticket_use.max_transfers.is_none() || ticket_use.max_transfers == Some(0))
+                && ticket_use.boarding_time_limit.is_none()
+                && ticket_use.alighting_time_limit.is_none()
+            {
+                return true;
+            }
+        }
+        false
+    }
+
+    let mut prices_v1: Vec<PriceV1> = vec![];
+    let mut fares_v1: Vec<FareV1> = vec![];
+
+    // Conversion of OD fares on specific lines
+
+    for ticket_use_restriction in ticket_use_restrictions
+        .values()
+        .filter(|r| r.restriction_type == RestrictionType::OriginDestination)
+        .filter(|r| has_no_transfer(&ticket_uses, &r.ticket_use_id))
+    {
+        let ticket_use_id = &ticket_use_restriction.ticket_use_id;
+
+        let ticket = if let Some(ticket_use) = ticket_uses.get(ticket_use_id) {
+            if let Some(ticket) = tickets.get(&ticket_use.ticket_id) {
+                ticket
+            } else {
+                info!("Failed to find Ticket {:?}", ticket_use.ticket_id);
+                continue;
+            }
+        } else {
+            info!("Failed to find TicketUse {:?}", ticket_use_id);
+            continue;
+        };
+
+        let perimeters: Vec<_> = ticket_use_perimeters
+            .into_iter()
+            .filter(|(_, ticket_use_perimeter)| {
+                &ticket_use_perimeter.ticket_use_id == ticket_use_id
+                    && ticket_use_perimeter.object_type == ObjectType::Line
+                    && ticket_use_perimeter.perimeter_action == PerimeterAction::Included
+            })
+            .map(|(_, p)| p)
+            .collect();
+
+        if perimeters.is_empty() {
+            info!(
+                "Failed to find TicketUsePerimeter for TicketUse {:?}",
+                ticket_use_id
+            );
+            continue;
+        }
+
+        let prices: Vec<_> = ticket_prices
+            .into_iter()
+            .filter(|(_, ticket_price)| {
+                &ticket_price.ticket_id == &ticket.id && ticket_price.currency == "EUR".to_string()
+            })
+            .map(|(_, p)| p)
+            .collect();
+
+        if prices.is_empty() {
+            info!("Failed to find TicketPrice for Ticket {:?}", ticket.id);
+            continue;
+        }
+
+        for price in prices {
+            let cents_price = if let Some(cents_price) =
+                (price.price * Decimal::from(100)).round_dp(0).to_u32()
+            {
+                cents_price
+            } else {
+                0
```

Comment:
Do we really want to put a `0` in case we're not able to convert whatever comes in into a number? Let's discuss it.